File: Selenium/Activities/Activity17.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;


public class Activity17 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Title of webpage:" + driver.getTitle());
        Select singleSelect= new Select(driver.findElement(By.id("single-select")));
        singleSelect.selectByVisibleText("Option 2");
        System.out.println("Selected the second option using visible text:"+singleSelect.getFirstSelectedOption().getText());
        singleSelect.selectByIndex(3);
        System.out.println("Selected the third option using index:"+singleSelect.getFirstSelectedOption().getText());
        singleSelect.selectByValue("4");
        System.out.println("Selected the fourth option using value:"+singleSelect.getFirstSelectedOption().getText());
        System.out.println("All items from select dropdown:");
        for(WebElement selectOption:singleSelect.getOptions()){
            System.out.println(selectOption.getText());

        }
        driver.close();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/selects")
    wait = WebDriverWait(driver, 10)
    print("Title of webpage:" + driver.title)
    singleSelect = Select(driver.find_element(By.ID, "single-select"))
    singleSelect.select_by_visible_text("Option 2")
    print("Selected the second option using visible text:" + singleSelect.first_selected_option.text)
    singleSelect.select_by_index(3)
    print("Selected the third option using index:" + singleSelect.first_selected_option.text)
    singleSelect.select_by_value("4")
    print("Selected the fourth option using value:" + singleSelect.first_selected_option.text)
    print("All items from select dropdown:")
    for selectOption in singleSelect.options:
        print(selectOption.text)
    driver.close()
